web/src/main/java/com/wilma/web/controller/web/portal/EducatorPortalController.java

```diff
@@ -29,31 +46,147 @@ public String dashboard(Model model) {
+        var categoryName = reply.getPost().getCategory().getName();
+        model.addAllAttributes(Map.of(
+                "currentPage", "forum",
+                "menuElements", UserConfiguration.educatorMenuElements,
+                "postId", postId,
+                "categoryName", categoryName,
+                "postsByCategory", forumService.getPostByCategoryName(categoryName),
+                "repliesForPosts", forumService.getPostRepliesByCategory(categoryName),
+                "reply", replyDTO));
+
+        log.info("Reply added: "+ reply);
+        return new RedirectView("/educator/forum-thread?category="+ categoryName);
+    }
+    @PostMapping("/create-post")
+    public RedirectView createPost(@ModelAttribute PostDTO postDTO, Model model){
+        var newPost = forumService.addPostFromDTO(postDTO);
+        var categoryName = newPost.getCategory().getName();
+        model.addAllAttributes(Map.of(
+                "currentPage", "forum",
+                "menuElements", UserConfiguration.educatorMenuElements,
+                "availableCategories", categoryService.findAll(),
+                "availableTags", tagService.findAll(),
+                "categoryName", categoryName,
+                "postsByCategory", forumService.getPostByCategoryName(categoryName),
+                "repliesForPosts", forumService.getPostRepliesByCategory(categoryName),
+                "post", postDTO));
+
+        log.info("Post created from DTO: "+ newPost);
+        return new RedirectView("/educator/forum");
+    }
+
+
+    @GetMapping("/forum-thread")
+    public String forumThread(@RequestParam String category, Model model) {
+        model.addAllAttributes(Map.of(
+                "currentPage", "forum",
+                "menuElements", UserConfiguration.educatorMenuElements,
+                "categoryName", category,
+                "postsByCategory", forumService.getPostByCategoryName(category),
+                "repliesForPosts", forumService.getPostRepliesByCategory(category)));
+        return "/educator/forum/forum-thread";
+    }
+
+    @GetMapping("/delete-post/{postId}")
+    public RedirectView deleteForumPost(@PathVariable Integer postId, @RequestParam String category, Model model){
+        forumService.deleteById(postId);
+        model.addAllAttributes(Map.of(
+                "currentPage", "forum",
+                "menuElements", UserConfiguration.educatorMenuElements,
+                "categoryName", category,
+                "postsByCategory", forumService.getPostByCategoryName(category),
+                "repliesForPosts", forumService.getPostRepliesByCategory(category)));
+        return new RedirectView("/educator/forum-thread?category=" + category);
+    }
+    //endregion
+
+    //region Expressions of interest
+    @GetMapping("/expressions-of-interest")
+    public String expressionsOfInterest(Model model) {
+        model.addAllAttributes(Map.of(
+                "currentPage", "Expressions Of Interest",
```

Comment:
Menu item not being selected
```suggestion
"currentPage", "expressions-of-interest",
```